### PR TITLE
Set up VS templates using dotnet template engine

### PR DESCRIPTION
**Pull request by Jjagg**
This sets up VSIX (VS extension) generation from the dotnet/templating templates.
Will also automate publishing the extension to the Visual Studio Marketplace.
Draft because this stuff needs to be done in the CAKE build scripts.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -203,17 +220,41 @@ Task("PackVSTemplates")
...
+});
+Task("PublishVSTemplates")
+    .IsDependentOn("PackVSTemplates")
+    .WithCriteria(() => IsRunningOnWindows())
+    .Does(() =>
+{
+    if (!GetVsixPublisher(out var vsixPublisher))
+        throw new Exception("Did not find VsixPublisher.exe.");
+
+    var pat = EnvironmentVariable("AZURE_DEVOPS_MARKETPLACE_PAT");
```
Comment (Jjagg) [inline build.cake:245]: @tomspilman I have the Personal Access Token (PAT). It'll need to be set as an environment variable in the teamcity build settings.

**Comment by harry-cpp** [comment]
Whats the status on this?

File: build.cake [hunk 2]
```diff
@@ -241,14 +282,16 @@ Task("BuildAll")
...
-    .IsDependentOn("PackVSTemplates")
     .IsDependentOn("PackVSMacTemplates");
 
 Task("Test")
     .IsDependentOn("TestWindowsDX")
     .IsDependentOn("TestDesktopGL")
     .IsDependentOn("TestTools");
 
+Task("PublishStable")
+    .IsDependentOn("PublishVSTemplates");
```
Comment (Jjagg) [inline build.cake:293]: @harry-cpp I'm not sure when we want to publish new templates. Every time the develop branch changes? Apart from this, this PR is ready.

Comment (harry-cpp) [reply build.cake:293]: I think so, yea.

Comment (harry-cpp) [reply build.cake:293]: Wait a second, are you talking about publishing vsix to teamcity or marketplace, the later should be done only on releases.

Comment (tomspilman) [reply build.cake:293]: Yeah seems better for this to be a Teamcity job to publish the template and not a cake script thing.

Comment (harry-cpp) [reply build.cake:293]: @Jjagg Any chance you can update this?

**Pull request merged**